### ISSUE TITLE
Memory leak during inference 

### ISSUE THREAD

**Issue by shameem198** [open-mmlab/mmdeploy#309]
Hi,

I have converted **cascade mask rcnn detection** model from **PyTorch** to **ONNX** using **MMDeploy**. During the load test, I noticed that **RAM consumption is increasing** with respect to number of inferences. I Followed official documentation for environment setup.

**platform = Linux-x86_64
python = 3.7
pytorch = 1.8.0
torchvision = 0.9.0
mmcv-full = 1.4.0
mmdet = 2.20.0
onnxruntime = 1.8.1
device = cpu**

and using given script for inference _https://github.com/open-mmlab/mmdeploy/blob/master/mmdeploy/apis/inference.py_ 

```
        deploy_cfg, model_cfg = load_config(deploy_cfg, model_cfg)
        task_processor = build_task_processor(model_cfg, deploy_cfg, 'cpu')
        model = task_processor.init_backend_model([backend_files])
        input_shape = get_input_shape(deploy_cfg)

        model_inputs, _= task_processor.create_input(img, input_shape)
        with torch.no_grad():
            result = model(**model_inputs)


```

I have already tried with latest versions of above packages, garbage collection and deleted unnecessary variables after inference. It didn't help to reduce memory leak. How do we fix this issue?

![memory](https://user-images.githubusercontent.com/45461737/161519183-47d5b49f-bab7-4d12-89e6-ab4adddfd70b.jpg)

**Comment by grimoire** [comment]
Hi, would you mind telling me how to do profile the memory usage?

**Comment by shameem198** [comment]
@grimoire I have used psutil library, also cross checked with htop command
`
import os
import psutil
pid = os.getpid()
total_memory = psutil.virtual_memory().total/(1024*1024) #(in MB)
process = psutil.Process(pid)
memor_utilization=f"{float(process.memory_percent()/100)*(total_memory):.2f}

`

**Comment by shameem198** [comment]
@grimoire is there any way to keep memory usage stable in above scenario?

**Comment by shameem198** [comment]
@grimoire I disabled **enable_cpu_mem_arena** and **enable_mem_pattern**  in ONNXRuntimeSession, which helped to keep memory and cpu stable, but there is a slight increment in inference time. There's a trade-off between these parameters and inference time.
